compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -86,18 +86,55 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 /// Intermediate format to store the hir_id pointing to the use that resulted in the
 /// corresponding place being captured and a String which contains the captured value's
 /// name (i.e: a.b.c)
-type CapturesInfo = (Option<hir::HirId>, String);
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+enum CapturesInfo {
```

Comment:
Is this only used for migration? If so I'd name is UpvarMigrationInfo for two reasons:
1. Upvar indicates we are taling about root variable, and migration helps with inteded purpose.
2. CaptureInfo very similarly named struct stores information about the information that we collect when run the analysis algorithm.

Reply:
Yes. Good thought.